### DISCUSSION TITLE
python3-libdnf to python3-libdnf5 migration

### DISCUSSION THREAD

**Discussion by JGoutin** [rpm-software-management/dnf5#1867] [Q&A] [answered]
Hello,

I have [the following Python script](https://github.com/JGoutin/ansible_home/blob/master/roles/squid/files/squid_dnf_mirrors.py) working with the legacy libdnf.

I want to migrate it to use libdnf5 instead ([Current attempt here](https://github.com/JGoutin/ansible_home/blob/squid_dnf_mirrors_dnf5/roles/squid/files/squid_dnf_mirrors.py))

The objective is to list all installed repositories and then list all mirrors for these repositories.

I find that there is a `get_mirrors()` method on the `Repo` class [here](https://github.com/rpm-software-management/dnf5/blob/708604c8ac6d1e9fa48dad97337e27423a880a5a/include/libdnf5/repo/repo.hpp#L302). (Note that this method is not listed in the [doc](https://dnf5.readthedocs.io/en/stable/api/python/libdnf5_repo.html))

But, I don't find the way to get a list of installed `Repo`  instances (Or IDs that I can use with `RepoSack.create_repo`).

Thanks.

**Comment by kontura** [comment, marked as answer]
This prints ids and mirrors for all installed and enabled repositories. 
```python
import libdnf5

base = libdnf5.base.Base()
base.setup()
repo_sack = base.get_repo_sack()
repo_sack.create_repos_from_system_configuration()
repo_sack.load_repos(libdnf5.repo.Repo.Type_AVAILABLE)

query = libdnf5.repo.RepoQuery(base)
query.filter_enabled(True)

for repo in query:
    print(repo.get_id())
    print(repo.get_mirrors())
```
You can check out some details about the loading of configured repositories in https://dnf5.readthedocs.io/en/latest/tutorial/bindings/python3/repos.html.

Note that the `libdnf5.repo.RepoQuery` is not connected to the `repoquery` command on the CLI, it is a query on all known repositories (`Repo` objects, not packages contained in those repositories).

> (Note that this method is not listed in the [doc](https://dnf5.readthedocs.io/en/stable/api/python/libdnf5_repo.html))

Yea, the docs miss the `libdnf5::repo::Repo` object completely. Unfortunately a lot of the API is not currently exposed in the docs.

**Reply by JGoutin** [reply]
That's perfect. Thank you!
